connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoader.java

```diff
@@ -187,17 +192,24 @@ private static PluginClassLoader newPluginClassLoader(
         );
     }
 
-    private <T> void addPlugins(Collection<PluginDesc<T>> plugins, ClassLoader loader) {
+    //visible for testing
+    <T> void addPlugins(Collection<PluginDesc<T>> plugins, ClassLoader loader) {
         for (PluginDesc<T> plugin : plugins) {
             String pluginClassName = plugin.className();
             SortedMap<PluginDesc<?>, ClassLoader> inner = pluginLoaders.get(pluginClassName);
+            boolean pluginConflict = false;
             if (inner == null) {
                 inner = new TreeMap<>();
                 pluginLoaders.put(pluginClassName, inner);
                 // TODO: once versioning is enabled this line should be moved outside this if branch
                 log.info("Added plugin '{}'", pluginClassName);
+            } else {
+                pluginConflict = true;
             }
             inner.put(plugin, loader);
+            if (pluginConflict) {
+                log.error("Detected multiple copies of plugin '{}', one of these will be used '{}'", pluginClassName, inner.keySet());
+            }
```

Comment:
Do we need the `pluginConflict` field, or can this be moved into the `else` branch above?

Reply:
we can't move in else because we need to report all versions and they are added outside the if/else block. Also, I cant use size() check because sometimes the same plugin with the same version exists in 2 places.

Reply:
Oh, gotcha--in that case, should we do a check somewhere else, since this will be triggered potentially multiple times for a single plugin? For example, if there are three copies of a connector, the warning will be logged twice right now, with different values for `inner.keySet()` each time.

Also, it may also help to log exactly which one we're going to use either instead of or in addition to the complete set of discovered versions of the duplicated plugin.

Reply:
I had discussed that case with Randall and you are right detecting and logging only once would need the change to inspect the map after all plugins are loaded but soon we will be working on dynamically loading plugins at runtime to improve the worker startup time so it would be a moot point to do it that way.  This is why it's ok to see the same statement twice as it would be rare for someone to have 3 copies of the same plugin.

sure let me make the change to log which one will be used as of the time when statement is logged.

Reply:
Mmmm, I'm not sure we should be making decisions here based on dynamic plugin loading for two reasons:

1. This change can be backported to older versions of Connect, which will never have that feature.
2. It's unclear exactly what the mechanism for dynamic plugin loading will be, and it's possible that a re-scan of all known plugins after loading has taken place (either the initial start load or a subsequent dynamic load at runtime) could still be beneficial

Also, it's actually not that uncommon for 3+ copies of the same plugin to appear on the plugin path for a worker. For example, some connectors come packaged directly with converters; all it takes is at least two such connectors and a separately-installed copy of that converter to lead to that number of copies.

Reply:
@C0urante you are right, I reworked the PR after your comments, Could you please check now?